### PR TITLE
Use Version Based Profile to Compile Groovy

### PR DESCRIPTION
**Pull request by alzimmermsft**
Updates the Blob and Queue encryption packages to use version based profiles to compile the Groovy tests to support Java 8 and Java 9+.

### CODE REVIEW

File: sdk/storage/azure-storage-blob-cryptography/pom.xml [hunk 1]
```diff
@@ -72,32 +72,148 @@
...
+              </dependency>
+            </dependencies>
+          </plugin>
+
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-surefire-plugin</artifactId>
+            <version>${maven-surefire-plugin.version}</version>
+            <configuration>
+              <argLine>
```
Comment (srnagar) [inline sdk/storage/azure-storage-blob-cryptography/pom.xml]: You can remove this argLine property until you have source code that requires these `exports` and `opens`.

File: sdk/storage/azure-storage-queue-cryptography/pom.xml [hunk 2]
```diff
@@ -72,32 +72,148 @@
...
+              </dependency>
+            </dependencies>
+          </plugin>
+
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-surefire-plugin</artifactId>
+            <version>${maven-surefire-plugin.version}</version>
+            <configuration>
+              <argLine>
```
Comment (srnagar) [inline sdk/storage/azure-storage-queue-cryptography/pom.xml]: same here

Comment (alzimmermsft) [reply sdk/storage/azure-storage-queue-cryptography/pom.xml]: Removed

File: sdk/storage/azure-storage-blob-cryptography/pom.xml (hunk 1 above)
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob-cryptography/pom.xml]: Removed

**Pull request merged**